src/client/pkg/grpcutil/pool.go

```diff
@@ -52,6 +118,7 @@ func (p *Pool) Put(conn *grpc.ClientConn) error {
 // Close closes all connections stored in the pool, it returns an error if any
 // of the calls to Close error.
 func (p *Pool) Close() error {
+	p.endpointsWatch.Stop()
 	var retErr error
 	for conn := range p.conns {
 		if err := conn.Close(); err != nil {
```

Comment:
My intention was to leave pool-related optimizations to future PRs.  Here I'm just doing the minimal to get around the bug that a pod cannot talk to itself via service IP.